requre/helpers/files.py

```diff
@@ -59,24 +129,50 @@ def _copy_logic(cls, cassette: Cassette, pathname: str, keys: list) -> None:
             try:
                 artifact_name = os.path.basename(pathname)
                 artifact_path = os.path.dirname(pathname)
+                if cls.store_optimalization_switch:
+                    realpath_file_location = os.path.realpath(pathname)
+                    if os.path.isdir(pathname):
+                        try:
+                            # try to use git if possible to get sha otherwise use md5
+                            # git files constains some changing timestamps and similar
+                            # so it is not able to match same git dir
+                            import git
+
+                            try:
+                                file_name = (
+                                    "gitsha:" + git.Repo(pathname).head.object.hexsha
+                                )
+                            except git.InvalidGitRepositoryError:
+                                file_name = "md5dir:" + md5_dir(realpath_file_location)
+                        except ImportError:
+                            file_name = "md5dir:" + md5_dir(realpath_file_location)
+                    else:
+                        file_name = "md5file:" + md5_file(realpath_file_location)
+                else:
+                    file_name = "timestamp:" + str(time.time())
```

Comment:
What will be the workflow when regeneratioin?

How do we treat removals? Can we do something about that?

I am worried that it can leave a lot of unused files around.